File: app/src/main/java/com/BlizzardArmory/warcraft/talents/Specialization.java
package com.BlizzardArmory.warcraft.talents;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class Specialization {

    @SerializedName("specialization")
    @Expose
    private SpecializationName specialization;
    @SerializedName("talents")
    @Expose
    private List<Talent> talents = null;
    @SerializedName("glyphs")
    @Expose
    private List<Glyph> glyphs = null;
    @SerializedName("pvp_talent_slots")
    @Expose
    private List<PvpTalentSlot> pvpTalentSlots = null;

    public SpecializationName getSpecialization() {
        return specialization;
    }

    public void setSpecialization(SpecializationName specialization) {
        this.specialization = specialization;
    }

    public List<Talent> getTalents() {
        return talents;
    }

    public void setTalents(List<Talent> talents) {
        this.talents = talents;
    }

    public List<Glyph> getGlyphs() {
        return glyphs;
    }

    public void setGlyphs(List<Glyph> glyphs) {
        this.glyphs = glyphs;
    }

    public List<PvpTalentSlot> getPvpTalentSlots() {
        return pvpTalentSlots;
    }

    public void setPvpTalentSlots(List<PvpTalentSlot> pvpTalentSlots) {
        this.pvpTalentSlots = pvpTalentSlots;
    }

}

File: app/src/main/java/com/BlizzardArmory/model/warcraft/talents/Specialization.kt
package com.BlizzardArmory.model.warcraft.talents

import com.google.gson.annotations.SerializedName


/**
 * The type Specialization.
 */
data class Specialization(
        @SerializedName("specialization")
        var specialization: SpecializationName,

        @SerializedName("talents")
        var talents: List<Talent>,

        @SerializedName("glyphs")
        var glyphs: List<Glyph>,

        @SerializedName("pvp_talent_slots")
        var pvpTalentSlots: List<PvpTalentSlot>

)